src/sentry/web/frontend/integration_extension_configuration.py

```diff
@@ -22,9 +22,10 @@ def _dialog_success(self, _org_integration):
         if "next" in self.request.GET:
             param_string = "?%s" % urlencode({"next": self.request.GET["next"]})
 
-        redirect_uri = (
-            f"/settings/{org_slug}/integrations/{provider}/{integration_id}/{param_string}"
+        redirect_uri = self.organization.absolute_url(
+            f"/settings/{org_slug}/integrations/{provider}/{integration_id}/"
         )
+        redirect_uri = f"{redirect_uri.rstrip('/')}/{param_string}"
```

Comment:
```suggestion
        redirect_uri = self.organization.absolute_url(
            f"/settings/{org_slug}/integrations/{provider}/{integration_id}/"
            query=param_string
        )
```

You should be able to do this as well and save some string operations.